python/lsst/meas/algorithms/matcherSourceSelector.py

```diff
@@ -162,4 +162,68 @@ def _isUsable(self, source):
             and not source.get(self.fluxFlagKey) \
             and self._goodSN(source)
 
+
+class MatcherPessimisticSourceSelectorTask(MatcherSourceSelectorTask):
+    """
+    !Select sources that are useful for matching.
+
+    Good matching sources have high signal/noise, are non-blended. They need not
+    be PSF sources, just have reliable centroids. This inherited class adds
+    the removal of saturated, interpolated, and edge_key objects to the set of
+    bad flags. It is a temporary addition designed preserve the source selction
+    used in matchOptimisticB. Once matchPessimisticB is adopted as the default
+    source selector the class will be removed and the saturated, interpoalted, and
+    edge_key flags will be added to the matcherSourceSelector class.
+
+    TODO: Once DM-10399 is complete an RFC will be filed to make matchPessimisticB
+    the default matcher this class will replace matcherSourceSelector with this source
+    selector resulting in only one matcherSourceSeletor.
```

Comment:
We can't get rid of the old source selector until we get rid of the old matcher. I suspect that will take a long time during which both matchers will be available. That sounds like two tickets: first make the new matcher the default, then months later (once folks sign of on it) delete the old matcher. These selectors won't be clean until we do the latter. DM-10399 is not the ticket for either of these. So I'm suggesting you file a new ticket and reference it here.

Better yet, just give this matcher a nice name that we can use for the long term and keep the name. That avoids technical debt and the need to reference any ticket.